### PR TITLE
Do not share staging directory between tests

### PR DESCRIPTION
**Pull request by losipiuk**

### CODE REVIEW

File: plugin/trino-hive/src/test/java/io/trino/plugin/hive/AbstractTestHive.java [hunk 1]
```diff
@@ -857,6 +859,8 @@ protected HiveConfig getHiveConfig()
     {
         return new HiveConfig()
                 .setMaxOpenSortFiles(10)
+                // Use separate staging directory for each test class to prevent intermittent failures coming from test parallelism
+                .setTemporaryStagingDirectoryPath("/tmp/presto-${USER}-" + UPPER_CAMEL.converterTo(LOWER_HYPHEN).convert(this.getClass().getSimpleName()))
```
Comment (sopel39) [inline plugin/trino-hive/src/test/java/io/trino/plugin/hive/AbstractTestHive.java:863]:
```
UPPER_CAMEL.converterTo(LOWER_HYPHEN).convert(this.getClass().getSimpleName()
```
seems complex. Just use some static string here?

Comment (sopel39) [inline plugin/trino-hive/src/test/java/io/trino/plugin/hive/AbstractTestHive.java:863]: In `AbstractTestHive` run with different users

**Review by sopel39 — APPROVED** [review]

File: plugin/trino-hive/src/test/java/io/trino/plugin/hive/AbstractTestHive.java (hunk 1 above)
Comment (losipiuk) [reply plugin/trino-hive/src/test/java/io/trino/plugin/hive/AbstractTestHive.java:863]: Cannot be static. It needs to be different for each test class.

File: plugin/trino-hive/src/test/java/io/trino/plugin/hive/AbstractTestHive.java [hunk 2]
```diff
@@ -857,6 +859,8 @@ protected HiveConfig getHiveConfig()
     {
         return new HiveConfig()
                 .setMaxOpenSortFiles(10)
+                // Use separate staging directory for each test class to prevent intermittent failures coming from test parallelism
+                .setTemporaryStagingDirectoryPath("/tmp/trino-${USER}-" + UPPER_CAMEL.converterTo(LOWER_HYPHEN).convert(this.getClass().getSimpleName()))
```
Comment (findepi) [inline plugin/trino-hive/src/test/java/io/trino/plugin/hive/AbstractTestHive.java:863]: Can this just be randomized, e.g. UUID based?

Also, for tmp local resources, we should also

```
deleteRecursively(..., ALLOW_INSECURE)
```

**Comment by losipiuk** [comment]
ac @findepi @sopel39

**Review by findepi — APPROVED** [review]

**Comment by losipiuk** [comment]
CI fixed already on master with https://github.com/trinodb/trino/pull/7086

**Pull request merged**